lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -311,32 +364,38 @@ def _user_certificate_info(self, user, context, course_grade, whitelisted_user_i
             course_grade.letter_grade,
             context.course.grade_cutoffs,
             user.profile.allow_certificate,
-            user.id in whitelisted_user_ids,
+            is_whitelisted,
         )
         return certificate_info
 
     def _rows_for_users(self, context, users):
         """
         Returns a list of rows for the given users for this report.
         """
-        certificate_whitelist = CertificateWhitelist.objects.filter(course_id=context.course_id, whitelist=True)
-        whitelisted_user_ids = [entry.user_id for entry in certificate_whitelist]
```

Comment:
This is now handled in bulk via [_CertificateBulkContext](https://github.com/edx/edx-platform/pull/15036/files#diff-7683b9bc4a5398602e597d135a5af15bR124).